featurehub-javascript-client-sdk/app/client_feature_repository.ts

```diff
@@ -246,8 +246,8 @@ export class ClientFeatureRepository implements InternalFeatureRepository {
     if (features && features.length > 0) {
       features.forEach((f) => {
         const existingFeature = this.features.get(f.key);
-        if (!existingFeature || (existingFeature.getKey()
-          && f.version > existingFeature.getFeatureState().version)) {
+        if (!existingFeature || !existingFeature.exists || (existingFeature.getKey()
```

Comment:
how can it be !existingFeature.exists if this.feature.get(f.key) has returned something?

Reply:
because you added a listener before your key existed. So  it always creates a FeatureBaseHolder to hold your listener and when the feature turns  up, it passes that state onto the real feature. So the feature is in the map, but doesn't actually exist in the backend list of features.